### PR TITLE
Background Configurator: #411 (Add possibility to set background color|image|video)

### PR DESCRIPTION
**Pull request by Diegiwg**
Element to configure the background of the application.

There are three options:

1. Color: set a solid color for the app's background.
2. Image: define an image, and its visualization behavior, for the background of the app.
3. Set a video (.mp4) to run in a loop and muted, as the background of the app.

> I decided to create a new element, because using ui.colors to also manage the app's background would be too much responsibility for a single element.

### CODE REVIEW

File: nicegui/elements/background.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+from typing import Literal, Optional
```
Comment (rodja) [inline nicegui/elements/background.py:1]: Literal is not part of Python 3.7 and must be imported via 

```py
from typing_extensions import Literal
```

File: website/reference.py [hunk 2]
```diff
@@ -542,6 +542,21 @@ def colors_example():
...
+        # ui.background(
+        #     image_src=IMAGE_LINK,
+        #     image_repeat="no-repeat",
+        # )
+
+        # ui.background(
+        #     video_src=VIDEO_LINK
+        # )
+
+        ui.label("Unfortunately, it is not possible to demonstrate here the operation of the element.")
```
Comment (rodja) [inline website/reference.py:558]: Maybe we can use a demo like this:

```py
def set_background(e: events.ValueChangeEventArguments):
    brighness = e.value * 100
    ui.background(color=f'hsl(0, 0%, {brighness}%)')

ui.label('use the slider to set background color')
ui.slider(min=0, max=1, step=0.01, on_change=set_background)
```

**Review by falkoschindler — COMMENTED** [review]
Thanks for this PR, @Diegiwg!

I like the idea of introducing a dedicated background element. However, I have some concerns about this implementation:

- It is pretty specialized. For more CSS properties we need to change the JavaScript component every time.
- You can't set color _and_ video.
- Although one might want to place a video fullscreen "on the background", it is technically simply another element fixed to the screen. This can already be done with `ui.video`.
- You can't have a video that scrolls with the background.

Long story short: It is a rather complicated implementation for a comparatively limited feature set. Let us continue discussing an alternative implementation I have in mind on #411.

**Comment by Diegiwg** [comment]
I hadn't thought of using the .class to manipulate the background. I believe that your implementation has become much closer to the Nicegui API.

**Comment by rodja** [comment]
I gave both PR's some thought. While #411 is more generic it tries to mimic the behaviour of other elements. But it's a singleton (it will not create a new element like the other ones but just gives you access to the body tag). Thereby it behaves quite strange. For example

```py
with ui.background():
   ui.label('this is not shown')
```
I think deriving from ui.element is not the best approach. This thing is not an element. In my opinion it should have an different API.

Also the name is not perfect in my opinion. You can use it to set padding or font colors for the whole document which is nice but does not relate well to "background".

**Comment by rodja** [comment]
We have some other topics which may relate to this:

- existing functions which access/manipulate the body-tag
   - `ui.colors`
   - `ui.add_body_html`
- simplified styling (#117, #364)

I still can't wrap my head around this all.

**Comment by rodja** [comment]
I've put my comments into #411, because they are more generic: https://github.com/zauberzeug/nicegui/issues/411#issuecomment-1457879985

**Comment by falkoschindler** [comment]
I'll close this PR due to new ideas and insights about this topic (#411).
But thanks again for your effort, @Diegiwg!
